src/cms.c

```diff
@@ -0,0 +1,93 @@
+#include <stdlib.h>         // malloc
+#include <string.h>         // strlen
+#include <stdio.h>          // printf
+#include <math.h>           // q, ceil
+#include <assert.h>         // assert
+
+#include "cms.h"
+
+CMSketch *NewCMSketch(size_t width, size_t depth) {
+    assert(width > 0);
+    assert(depth > 0);
+
+    CMSketch *cms = NULL;
+    if((cms = (CMSketch *)CMS_CALLOC(1, sizeof(CMSketch))) == NULL) {
```

Comment:
I see some mixed code about this. Since it isn't called much, it might be OK.
This is a good opportunity to get out of the bubble of Infinity to the real world :)